src/axom/primal/tests/primal_integral.cpp

```diff
@@ -231,20 +249,64 @@ TEST(primal_integral, evaluate_line_integral_vector)
                                   Point2D {0.0, -3.0},
                                   Point2D {1.0, -1.0}};
   Bezier para2_shift(paranodes2_shifted, 2);
-  axom::Array<Bezier> disconnected_parabola_shape({para1, para2_shift});
+
+  Bezier disconnected_parabola_edges[] = {para1, para2_shift};
+  CPolygon disconnected_parabola_shape(disconnected_parabola_edges, 2);
 
   EXPECT_NEAR(
-    evaluate_line_integral(disconnected_parabola_shape, area_field, npts),
+    evaluate_vector_line_integral(disconnected_parabola_shape, area_field, npts),
     11.0 / 3.0,
     abs_tol);
-  EXPECT_NEAR(
-    evaluate_line_integral(disconnected_parabola_shape, conservative_field, npts),
-    0.0,
-    abs_tol);
-  EXPECT_NEAR(
-    evaluate_line_integral(disconnected_parabola_shape, winding_field, npts),
-    0.75,
-    abs_tol);
+  EXPECT_NEAR(evaluate_vector_line_integral(disconnected_parabola_shape,
+                                            conservative_field,
+                                            npts),
+              0.0,
+              abs_tol);
+  EXPECT_NEAR(evaluate_vector_line_integral(disconnected_parabola_shape,
+                                            winding_field,
+                                            npts),
+              0.75,
+              abs_tol);
+}
+
+TEST(primal_rationalbezier, evaluate_integral_3D)
+{
+  using Point3D = primal::Point<double, 3>;
+  using Vector3D = primal::Vector<double, 3>;
+  using Bezier = primal::BezierCurve<double, 3>;
+  double abs_tol = 1e-10;
+
+  // Quadrature nodes. Should be sufficiently high to pass tests
+  int npts = 30;
+
+  const int order = 3;
+  Point3D data[order + 1] = {Point3D {0.6, 1.2, 1.0},
+                             Point3D {1.3, 1.6, 1.8},
+                             Point3D {2.9, 2.4, 2.3},
+                             Point3D {3.2, 3.5, 3.0}};
+  Bezier spatial_arc(data, 3);
+
+  auto const_integrand = [](Point3D /*x*/) -> double { return 1.0; };
+  auto transc_integrand = [](Point3D x) -> double {
+    return std::sin(x[0] * x[1] * x[2]);
+  };
+
+  auto vector_field = [](Point3D x) -> Vector3D {
+    return Vector3D({4 * x[1] * x[1], 8 * x[0] * x[1], 1.0});
+  };
+
+  // Test line integral on scalar domain
+  EXPECT_NEAR(evaluate_scalar_line_integral(spatial_arc, const_integrand, npts),
+              4.09193268998,
```

Comment:
How were these 3D integral values obtained?

Reply:
I computed these originally in desmos, but I did also compare them against a more robust scientific programming resource (mathematica) to verify them. I'll add a comment to clarify that.